### PR TITLE
Add support for thumbv7em with hard float

### PR DESCRIPTION
**Pull request by Codetector1374**

### CODE REVIEW

**Comment by hexdae** [comment]
Thanks for opening this. Having an example of cross-compilation for this target would also be awesome. I could help out with this if needed

File: rust/platform/triple_mappings.bzl [hunk 1]
```diff
@@ -188,11 +194,32 @@ _SYSTEM_TO_STDLIB_LINKFLAGS = {
...
 def cpu_arch_to_constraints(cpu_arch):
+    if cpu_arch not in _CPU_ARCH_TO_BUILTIN_PLAT_SUFFIX:
+        fail("CPU architecture \"{}\" is not supported by rules_rust".format(cpu_arch))
+
     plat_suffix = _CPU_ARCH_TO_BUILTIN_PLAT_SUFFIX[cpu_arch]
 
-    if not plat_suffix:
+    return ["@platforms//cpu:{}".format(plat_suffix)]
+
+def cpu_arch_constraints_from_triple(triple):
```
Comment (UebelAndre) [inline rust/platform/triple_mappings.bzl:204]: Why not update `cpu_arch_to_constraints` to match the pattern of `abi_to_constraints`? I don't particularly care which pattern is used but I would like there to be a consistent API.

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thanks! One question.

File: rust/platform/triple_mappings.bzl (hunk 1 above)
Comment (Codetector1374) [reply rust/platform/triple_mappings.bzl:204]: `cpu_arch_to_constraints` is also used in 
https://github.com/bazelbuild/rules_rust/blob/7f2dd433ac0f0fee6ee1ef90544a648640a66526/rust/platform/platform.bzl#L36
which I don't think it has access to the full abi / triple.

I can update `cpu_arch_constraints_from_triple` to match `abi_to_constraints` if that is what you are asking for.

Comment (UebelAndre) [reply rust/platform/triple_mappings.bzl:204]: Yeah, I would like that :)

Comment (Codetector1374) [reply rust/platform/triple_mappings.bzl]: Updated with the new interface.

**Review by UebelAndre — APPROVED** [review]
Thank you!

**Comment by UebelAndre** [comment]
Looks like you need a rebase

**Comment by Codetector1374** [comment]
done

**Pull request merged**